File: content-solutions/queue-deque/queue-dough-split.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] doughs, int limit) {
        Deque<Integer> line = new ArrayDeque<>();
        for (int w : doughs) line.offer(w);
        List<Integer> answer = new ArrayList<>();
        while (!line.isEmpty()) {
            int w = line.poll();
            if (w <= limit) {
                answer.add(w);
            } else {
                line.offer(w / 2);
                line.offer(w - w / 2);
            }
        }
        return answer;
    }
}

File: content-solutions/queue-deque/queue-dough-split.py
from collections import deque


def solution(doughs, limit):
    line = deque(doughs)
    answer = []
    while line:
        w = line.popleft()
        if w <= limit:
            answer.append(w)
        else:
            line.append(w // 2)
            line.append(w - w // 2)
    return answer
